src/components/CmsZone.vue

```diff
@@ -225,93 +285,56 @@ export default class CmsZone extends Vue {
-          await cmsClient.trackZone({ content, zoneId: this.zoneId });
-          this.$root.$emit(`cms.refresh.${this.zoneId}`);
-        }
-      );
-
-      this.$root.$once('router.change', (hash: string): void => {
-        const regex = new RegExp(trackOn);
-        if (regex.test(hash)) {
-          this.$root.$emit(`cms.track.${this.zoneId}`);
-        }
-      });
-      return;
+  private setupTracking(contents: Content[]): void {
+    for (let i = 0; i < contents.length; i++) {
+      const content = contents[i];
+      const trackOn = (content.extra || {}).track_on;
+      if (trackOn) {
+        this.setupDeferredTracking(content, trackOn);
+      } else {
+        this.setupIntersectionTracking(content, i);
+      }
     }
-
-    cmsClient.trackZone({ content, zoneId: this.zoneId });
   }
 
-  private trackScrollable(contents: Content[], scrollable: Element): void {
-    if (this.zoneType === 'carousel') {
-      let timeout: number;
-      const listener = (): void => {
-        window.clearTimeout(timeout);
-        timeout = window.setTimeout((): void => {
-          const contentElm = this.$el.querySelector('.slick-current');
-          if (contentElm && this.isContentVisible(contentElm, scrollable, percentVisible)) {
-            this.trackIndex(0);
-            scrollable.removeEventListener('scroll', listener);
-          }
-        }, durationVisibleToBeTrackedMs);
-      };
-      scrollable.addEventListener('scroll', listener);
-      this.scrollableListeners.push(listener);
-      Vue.nextTick(listener);
-      return;
-    }
+  private setupDeferredTracking(content: Content, trackOn: string) {
+    this.$root.$once(
+      `cms.track.${this.zoneId}`,
+      async (): Promise<void> => {
+        Vue.set(content, 'tracked', true);
+        await cmsClient.trackZone({ content, zoneId: this.zoneId });
+        this.$root.$emit(`cms.refresh.${this.zoneId}`);
+      }
+    );
 
-    contents.forEach((content, i): void => {
-      let timeout: number;
-      const listener = (): void => {
-        window.clearTimeout(timeout);
-        timeout = window.setTimeout((): void => {
-          const contentElm = this.$el.querySelector(`.cms-zone-content-${this.zoneId}-${i}`);
-          if (contentElm && this.isContentVisible(contentElm, scrollable, percentVisible)) {
-            this.trackIndex(i);
-            scrollable.removeEventListener('scroll', listener);
-          }
-        }, durationVisibleToBeTrackedMs);
-      };
-      scrollable.addEventListener('scroll', listener);
-      this.scrollableListeners.push(listener);
-      Vue.nextTick(listener);
+    this.$root.$once('router.change', (hash: string): void => {
+      const regex = new RegExp(trackOn);
+      if (regex.test(hash)) {
+        this.$root.$emit(`cms.track.${this.zoneId}`);
+      }
     });
   }
 
+  private setupIntersectionTracking(content: Content, index: number) {
+    const el = this.$el.querySelector(`.cms-zone-content-${this.zoneId}-${index}`)!;
```

Comment:
i think Vue prefers to use refs for this kind of thing

Reply:
I couldn't get it to work with refs, maybe they're set up too late?